python/lsst/ap/verify/pipeline_driver.py

```diff
@@ -175,16 +174,12 @@ def runApPipeGen3(workspace, parsedCmdLine, processes=1):
     pipelineArgs.extend(["--graph-fixup", "lsst.ap.verify.pipeline_driver._getExecOrder"])
 
     if not parsedCmdLine.skip_pipeline:
-        # CliRunner is an unsafe workaround for DM-26239
-        runner = click.testing.CliRunner()
+        # subprocess is an unsafe workaround for DM-26239
         # TODO: generalize this code in DM-26028
         # TODO: work off of workspace.workButler after DM-26239
-        results = runner.invoke(lsst.ctrl.mpexec.cli.pipetask.cli, pipelineArgs)
-        if results.exception:
-            raise RuntimeError("Pipeline failed.") from results.exception
-
+        result = subprocess.run(pipelineArgs, capture_output=False)
```

Comment:
in the previous implementation the pipetask run raised an exception on failure but this version returns an exit code (previously it raised but also returned an exit code). The docstring for this function does not say anything about what is returned. Can you update the docstring to be more explicit about intent here? If you still want it to raise I think you want `check=True` to be added here.

Reply:
Sorry, I'm confused. I know from testing that exceptions are propagated as-is, so the only behavior change there is the omission of one level of chaining. [Per the style guide](https://developer.lsst.io/python/numpydoc.html#py-docstring-raises), the exact exceptions raised by `runApPipeGen3` are not documented.

While `ctrl_mpexec` has been rewritten substantially since I first wrote this code, as far as I can tell there is still no documented guarantee that `pipetask run` or `CmdLineFwk.runPipeline` return any particular result. Therefore, formally I can't make any guarantee except that if there _is_ a meaningful return code, then it gets propagated.

Reply:
Okay. I'm also very confused. In the previous version of the code it had:

```
if results.exception:
    raise RuntimeError()
```
but in the current version of the code if pipetask fails the routine returns an exit code. Aren't those different behaviors? By default `subprocess.run()` does not raise an exception if the sub process returns bad exit status unless you also add `check=True`.

Reply:
The docstring doesn't tell me whether exit code is preferred over raising if there is a problem.